### PR TITLE
Fixes #17774: Duplicate category name error when saving a new user technique

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17774

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/TechniqueAcceptationDatetimeUpdater.scala [hunk 1]
```diff
@@ -117,9 +126,8 @@ class TechniqueAcceptationUpdater(
     @scala.annotation.tailrec
     def createCategories(names: List[CategoryInfo], parentCategory: (ActiveTechniqueCategoryId, String)): (ActiveTechniqueCategoryId, String) = {
       names match {
-        case Nil => parentCategory
-        case head::tail =>
-          val info = names.head
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/TechniqueAcceptationDatetimeUpdater.scala:122]: :upside_down_face:

Comment (fanf) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/TechniqueAcceptationDatetimeUpdater.scala:122]: well. We were young.

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**